components/output/mcp4461.rst

```diff
@@ -0,0 +1,161 @@
+  One of ``A``, ``B``, ``C``, ``D``, ``E``, ``F``, ``G`` or ``H``.
+- **initial_value** (**Optional**, float): Set initial wiper value, valid range is ``0.000 - 0.256``
+- **terminal_a** (**Optional**, bool): Set to ``false`` if terminal "A" shall be disabled on boot. Defaults to ``true``
+- **terminal_b** (**Optional**, bool): Set to ``false`` if terminal "B" shall be disabled on boot. Defaults to ``true``
+- **terminal_w** (**Optional**, bool): Set to ``false`` if terminal "W" shall be disabled on boot. Defaults to ``true``
+  
+- All other options from :ref:`Output <config-output>`.
+
+Tap count for 7/8bit digipot/rheostat devices is usually 100/257. Minimum setting for the output is 0.000, maximum is 0.256
+This example shows how to use this with simple numeric slider (0-256 non-decimal numbers shown)
+If initial_value configuration variable is not used, you can use
+
+.. code-block:: yaml
+
+    esphome:  
+      on_boot:
+        priority: 100
+        then:
+          - number.set:
+              id: digipot_volatile_0
+              value: !lambda |-
+                return id(digipot_wiper_0).get_wiper_level();
+
+Public Lambda functions:
+------------------------
+
+For mcp4461 component itself (not the wiper!):
+**********************************************
+
+- (uint16_t) get_eeprom_value(uint8_t location):
+    get current eeprom user data value of given eeprom-userdata-location (0-4) // 5x 9bits, max int val = 511 ! // ToDo improve eeprom usage
+- (void) set_eeprom_value(uint8_t location, uint16_t value):
+    set current eeprom user data value of given eeprom-userdata-location (0-4) // 5x 9bits, max int val = 511 !
+
+For the output (the wiper):
+***************************
+
+- (void) write_state(float state): // (protected, override, default frontend function)
+    expects value in range 0.00 to 0.256, higher values will be clamped to max value
+- (float) read_state(): // returns current wiper state, without (!) updating internal wiper state of output
+    returns current hw state of this wiper // NOT the internal state of the output
+- (float) update_state():
+    returns current value of this output (wiper) & updates internal state of output to this fetched value
+- (void) enable_wiper():
+    enable this output (wiper), only valid for volatile wipers (0-3)
+- (void) disable_wiper():
+    disable this output (wiper), only valid for volatile wipers (0-3)
+- (void) increase_wiper():
+    increase this output (wiper), only valid for volatile wipers (0-3)
+- (void) decrease_wiper():
+    decrease this output (wiper), only valid for volatile wipers (0-3)
+- (void) enable_terminal(char terminal):
+    enable terminal (one of ['a','b','w','h']) of this volatile wiper (0-3).
+    'h' is also used for general enable/disable of wiper
+- (void) disable_terminal(char terminal):
+    disable terminal (one of ['a','b','w','h']) of this volatile wiper (0-3).
+    'h' is also used for general enable/disable of wiper
```

Comment:
Move these comments to header file, doxygen will take care of them. Then mention that for lambda usage that user can read the API reference and add apiref link to both header files respectively. 

You can use doxygen special commands https://www.doxygen.nl/manual/commands.html with '@' symbol.

There are many ways to do it. Some examples:
https://github.com/esphome/esphome/blob/7f7f5d480b56e3d1ef8a13bee6ae676b407dd16f/esphome/components/remote_base/nec_protocol.h

https://github.com/esphome/esphome/blob/7f7f5d480b56e3d1ef8a13bee6ae676b407dd16f/esphome/core/helpers.h#L353